MetaMask/metamask-mobile
# [FIX] warning to users when BSC transactions would probably fail

**Description**
When we try to do a dapp transaction with Binance Smart Chain with a wallet with no funds or a transaction of a smart contract with some problem, we have an error on the RPC call eth_estimateGas and it was cancelling the transaction without giving feedback to the user.

**Proposed Solution**
Advertising the user that the transaction may fail because we were not able to estimate gas. The same solution as extension does it.
This PR to be tested and go to production also needs this [PR](https://github.com/MetaMask/metamask-mobile/pull/5200) of controllers to be merged.

**Screenshots/Recordings**
https://recordit.co/Q0xj57UqQN

Extension warning:
Before press the _I want to proceed anyway_
<img width=250 src="https://user-images.githubusercontent.com/46944231/192536616-d94740ce-6a52-4cdd-93d1-6ca66def3c07.png" />

After press the _I want to proceed anyway_
<img width=250 src="https://user-images.githubusercontent.com/46944231/192537016-9a7f3a00-8bac-4b04-bcfe-206e2b1db7ab.png" />


**Test Cases**
Switch network to Binance Smart Chain
Case1 (with a funded wallet and with a non funded wallet):
 * Go to chrome on your device
 * Browse https://opentarget.org/test/stupid/mm/wc/
 * Connect and do a transaction
 * The transaction modal will open with a warning saying gas couldn't be estimated (if you execute the tx will fail because no gas was estimated)
 * Go to Metamask test dapp on in app browser
 * Switch network to mainnet
 * Do an EIP 1559 tx
 * The modal should open without a warning
 Case2:
 * Browse on in app browser https://opentarget.org/test/stupid/mm/trans/
 * Connect the wallet
 * Do a transaction
 * The transaction modal will open with a warning saying gas couldn't be estimated (if you execute the tx will fail because no gas was estimated)

**Code Impact**
Medium will impact every transaction because it was added a new variable to the transaction object


**Issue**

Progresses #4646 #4621 #3049 

**Checklist**
